packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -774,3 +775,51 @@ export abstract class SharedObject<
 		telemetryContext?.set(this.telemetryContextPrefix, propertyName, prevTotal + incrementBy);
 	}
 }
+
+/**
+ * Defines a kind of shared object.
+ * Used in containers to register a shared object implementation, and to create new instances of a given type of shared object.
+ * @public
+ */
+export interface ISharedObjectKind<TSharedObject> {
+	/**
+	 * Get a factory which can be used by the Fluid Framework to programmatically instantiate shared objects within containers.
+	 * @remarks
+	 * The produced factory is intended for use with the FluidDataStoreRegistry and is used by the Fluid Framework to instantiate already existing Channels.
+	 * To create new shared objects use:
+	 *
+	 * - {@link @fluidframework/fluid-static#IFluidContainer.create} if using `@fluidframework/fluid-static`, for example via `@fluidframework/azure-client`.
+	 * - {@link ISharedObjectKind.create} if using a custom container definitions (and thus not using {@link @fluidframework/fluid-static#IFluidContainer}).
+	 *
+	 * @privateRemarks
+	 * TODO:
+	 * Many tests use this and can't use {@link ISharedObjectKind.create}.
+	 * The docs should make it clear why that's ok, and why {@link ISharedObjectKind.create} isn't in such a way that when reading non app code (like tests in this package)
+	 * someone can tell if the wrong one is being used without running it and seeing if it works.
+	 */
+	getFactory(): IChannelFactory<TSharedObject>;
+
+	/**
+	 * Create a shared object.
+	 * @param runtime - The data store runtime that the new shared object belongs to.
+	 * @param id - Optional name of the shared object.
+	 * @returns Newly created shared object.
+	 *
+	 * @example
+	 * To create a `SharedMap`, call the static create method:
```

Comment:
Nit: It occurs to me, since SharedMap is marked as deprecated, we should probably use SharedTree as the example case instead.

Reply:
Done.